tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+    escalated : bool
+        whether or not a case has been escalated. This field doesn't exist in 
+        the response until after a case has been escalated. True means the case
+        is escalated
+    case_creator : str
+        name of the user that opened the support case
+    create_time : str
+        timestamp of when the case was created
+    update_time : str
+        timestamp of the last update made to the case
+    priority : str
+        the current priority of the case, represented as S0, S1, S2, S3, or S4
+    state : str
+        the status of the support ticket. Can be NEW, IN_PROGRESS_GOOGLE_SUPPORT,
+        ACTION_REQUIRED, SOLUTION_PROVIDED, or CLOSED
+    comment_list : list
+        all public comments made on the case as strings. Comments are sorted with
+        newest comments at the top
+    """
+    
+    def __init__(self,caseobj):
+        """
+        Parameters
+        ----------
+        caseobj : json
+            json for an individual case
+
+        """
+        self.case_number = re.search('(?:cases/)([0-9]+)', caseobj['name'])[1]
+        self.resource_name = caseobj['name']
+        self.case_title = caseobj['displayName']
+        self.description = caseobj['description']
+        if 'escalated' in caseobj:
+            self.escalated = caseobj['escalated']
+        else:
+            self.escalated = False
+        self.case_creator = caseobj['creator']['displayName']
+        self.create_time = str(datetime.fromisoformat(
+            caseobj['createTime'].replace('Z', '+00:00')))
+        self.update_time = str(datetime.fromisoformat(
+            caseobj['updateTime'].replace('Z', '+00:00')))
+        self.priority = caseobj['severity'].replace('S', 'P')
+        self.state = caseobj['state']
+        req = support_service.cases().comments().list(parent=self.resource_name)
+        self.comment_list = req.execute().get('comments',[])
+
+
+        
+# Handle all calls to the support bot
+@app.route('/google-cloud-support', methods=['POST'])
+def gcp_support() -> Response:
+    """
+    Takes a user's slash command from Slack and executes it. Multiprocessing is used
+    on commands that modify the case to prevent Slack timeouts.
+
+    Parameters
+    ----------
+    request : Request
+        message and metadata that was submitted by Slack
+
+
+    Returns
+    -------
+    Response
+        tells Slack that the command was received and not to throw a timeout alert
+    200
+        HTTP 200 OK
+    """
+    
+    data = request.form
+    token = data.get('token')
+    channel_id = data.get('channel_id')
+    channel_name = data.get('channel_name')
+    user_id = data.get('user_id')
+    user_name = data.get('user_name')
+    user_inputs = data.get('text').split(' ', 1)
+    command = user_inputs[0]
+
+    if command == 'track-case':
+        try:
+            case = user_inputs[1]
+            track_case(channel_id, channel_name, case, user_id)
+        except Exception as e:
```

Comment:
What specific exception are you capturing here? Please narrow it down to the specific situations that you are trying to handle, this is too broad.